DataWorkerService/BackgroundServices/TournamentProcessorService.cs

```diff
@@ -49,45 +49,51 @@ protected override async Task DoWork(IServiceScope scope, CancellationToken stop
             scope.ServiceProvider.GetRequiredService<ITournamentProcessorResolver>();
 
         IEnumerable<Tournament> tournaments = await tournamentsRepository.GetNeedingProcessingAsync(_config.BatchSize);
+        var tasks = new List<Task>();
+
         foreach (Tournament tournament in tournaments)
         {
-            _stopwatch.Start();
-
-            logger.LogInformation(
-                "Processing starting [Id: {Id} | Processing Status: {Status} | Last Processing: {LastProcessing:MM/dd/yyyy}]",
-                tournament.Id,
-                tournament.ProcessingStatus,
-                tournament.LastProcessingDate
-            );
-
-            if (tournament.Matches.Count == 0)
-            {
-                tournament.VerificationStatus = VerificationStatus.Rejected;
-                tournament.RejectionReason = TournamentRejectionReason.IncompleteData;
-                tournament.ProcessingStatus = TournamentProcessingStatus.Done;
-                tournament.LastProcessingDate = DateTime.Now;
-
-                await context.SaveChangesAsync(stoppingToken);
-
-                logger.LogWarning("Skipping processing for tournament with no matches [Id: {Id}]", tournament.Id);
-                continue;
-            }
-
-            foreach (IProcessor<Tournament> processor in tournamentProcessorResolver.GetAll())
-            {
-                await processor.ProcessAsync(tournament, stoppingToken);
-            }
-
-            _stopwatch.Stop();
-            logger.LogInformation(
-                "Processing completed [Id: {Id} | Processing Status: {Status} | Elapsed: {Elapsed:mm\\:ss\\:fff}]",
-                tournament.Id,
-                tournament.ProcessingStatus,
-                _stopwatch.Elapsed
-            );
-            _stopwatch.Reset();
-
-            await context.SaveChangesAsync(stoppingToken);
+            tasks.Add(ProcessAsync(stoppingToken, tournament, tournamentProcessorResolver));
+        }
+
+        await Task.WhenAll(tasks);
+
+        await context.SaveChangesAsync(stoppingToken);
+    }
+
+    private async Task ProcessAsync(CancellationToken stoppingToken, Tournament tournament,
+        ITournamentProcessorResolver tournamentProcessorResolver)
+    {
+        var sw = new Stopwatch();
```

Comment:
Why create a new one when `_stopwatch` exists? This is seemingly not read from ever?

Reply:
Was my attempt at making it thread-safe, will need to think of something else I think.

Reply:
Actually in my head I can't see why a new stopwatch in each thread wouldn't work. But definitely can't use a shared one anymore.

Reply:
Opting to use a new format for this which I think looks much better.

![CleanShot 2024-12-19 at 15 12 58@2x](https://github.com/user-attachments/assets/e03d273d-1eaa-43b5-a981-be2b8e9a8939)

Reply:
Wouldn't it be easy to check for a 'no change' scenario on completion? Would be nice if there was a noticeable visual distinction between changing status and no change

Reply:
Yeah good idea